app/models/course/assessment/submission.rb

```diff
@@ -181,6 +181,16 @@ def current_answers
     answers.select(&:current_answer?).group_by(&:question_id).map { |pair| pair[1].first }
   end
 
+  # Loads the answer ids of the past answers of each question
+  def answer_history
+    answers.unscope(:order).order(created_at: :desc).group_by(&:question_id).map do |pair|
```

Comment:
Can suggest to use `pluck(:question_id, :id)` to not instantiate any ActiveRecord Object, a little more efficient.